src/frontend/src/optimizer/plan_node/logical_join.rs

```diff
@@ -334,8 +329,8 @@ impl LogicalJoin {
             .map(|index| l2i.try_map(*index))
             .chain(right_pk.iter().map(|index| r2i.try_map(*index)))
             .flatten()
-            .map(|index| i2o.try_map(index))
-            .collect::<Option<Vec<_>>>()
```

Comment:
I believe the previous implementation is incorrect? If the primary key from some side is not in the output, we should ignore it instead of `None.unwrap_or_default()` somewhere.